### PR TITLE
Adding setRow for distributed and sparse matrix types

### PR DESCRIPTION
**Pull request by rks171**
Description:
Allows the user to build a matrix row-by-row instead of one element
at a time, which can be much faster.

CASL Ticket # - N/A

### CODE REVIEW

File: src/MatrixTypes_Trilinos.f90 [hunk 1]
```diff
@@ -45,6 +45,8 @@ MODULE MatrixTypes_Trilinos
     INTEGER(SIK) :: ncol
     INTEGER(SIK),ALLOCATABLE :: jloc(:)
     REAL(SRK),ALLOCATABLE :: aloc(:)
+    LOGICAL(SBK) :: setByVal = .FALSE.
```
Comment (stimpsonsg) [inline src/MatrixTypes_Trilinos.f90]: beyond the scope of this, but need doxygen bits in this type definition

File: src/MatrixTypes_Trilinos.f90 [hunk 2]
```diff
@@ -210,6 +221,7 @@ SUBROUTINE set_TrilinosMatrixType(matrix,i,j,setval)
           ELSE
             IF(matrix%currow>0) THEN
               CALL ForPETRA_MatSet(matrix%A,matrix%currow,matrix%ncol,matrix%jloc,matrix%aloc)
+print *, matrix%currow, matrix%jloc, matrix%aloc
```
Comment (stimpsonsg) [inline src/MatrixTypes_Trilinos.f90]: debug statement that needs to be removed

File: src/MatrixTypes_Trilinos.f90 [hunk 3]
```diff
@@ -282,6 +321,7 @@ SUBROUTINE get_TrilinosMatrixType(matrix,i,j,getval)
 
         IF((i <= matrix%n) .AND. (j <= matrix%n) .AND. ((j > 0) .AND. (i > 0))) THEN
           CALL ForPETRA_MatGet(matrix%a,i,j,getval)
+print *, i, j, getval
```
Comment (stimpsonsg) [inline src/MatrixTypes_Trilinos.f90]: another one

File: src/MatrixTypes_Trilinos.f90 [hunk 4]
```diff
@@ -297,11 +337,14 @@ SUBROUTINE assemble_TrilinosMatrixType(thisMatrix,ierr)
...
       ierrc=0
       IF(.NOT.thisMatrix%isAssembled) THEN
-        CALL ForPETRA_MatSet(thisMatrix%A,thisMatrix%currow,thisMatrix%ncol,thisMatrix%jloc,thisMatrix%aloc)
-        thisMatrix%aloc=0.0_SRK
-        thisMatrix%jloc=0
-        thisMatrix%ncol=0
-        thisMatrix%currow=0
+        IF(thisMatrix%setByVal) THEN ! Add the last row
+          CALL ForPETRA_MatSet(thisMatrix%A,thisMatrix%currow,thisMatrix%ncol,thisMatrix%jloc,thisMatrix%aloc)
+          print *, thisMatrix%currow, thisMatrix%jloc, thisMatrix%aloc
```
Comment (stimpsonsg) [inline src/MatrixTypes_Trilinos.f90]: and here

File: src/MatrixTypes_Trilinos.f90 (hunk 1 above)
Comment (rks171) [reply src/MatrixTypes_Trilinos.f90]: Added it for all terms.

**Comment by rks171** [comment]
Removed all debug statements.  Seems like your comments disappeared when I edited the file online...

File: src/MatrixTypes_Trilinos.f90 (hunk 1 above)
Comment (stimpsonsg) [reply src/MatrixTypes_Trilinos.f90]: awesome, thanks

**Pull request merged**